File: MeichuHackaThon2024Model/Models/Path.cs
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel;
using MeichuHackaThon2024Model.Enums;
using Microsoft.EntityFrameworkCore;
using static Microsoft.EntityFrameworkCore.DbLoggerCategory.Database;


namespace MeichuHackaThon2024Model.Models
{
    public class Path
    {
        [Key]
        [DisplayName("流水號")]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public int Id { get; set; }

        [DisplayName("定位")]
        public string? Location { get; set; }

        [DisplayName("目的地")]
        public string? Destination { get; set; }

        [DisplayName("路線編號")]
        public int PathId { get; set; }

        [DisplayName("路線詳情編號")]
        public int PathDetailId { get; set; }

        [DisplayName("起點")]
        public int StartPointId { get; set; }  // 外鍵
        [ForeignKey("StartPointId")]
        public virtual PointInfo StartPoint { get; set; }  // 關聯

        [DisplayName("終點")]
        public int EndPointId { get; set; }  // 外鍵
        [ForeignKey("EndPointId")]
        public virtual PointInfo EndPoint { get; set; }  // 關聯


        [DisplayName("耗時(Min)")]
        public double CostTime { get; set; }

        [DisplayName("交通工具")]
        public int TransportationId { get; set; }  // 外鍵
        [ForeignKey("TransportationId")]
        public virtual Transportation Transportation { get; set; }  // 關聯

        [DisplayName("擁擠程度")]
        public CrowdingEnum Crowding { get; set; }

        // Updated to represent the one-to-many relationship
        [DisplayName("路線詳情")]
        public virtual ICollection<PathDetail> PathDetails { get; set; } = new List<PathDetail>();
    }
}

File: MeichuHackaThon2024Frontend/src/type/Path.ts
import type { PathDetail } from './PathDetail';

export type Path = {
  /**
   * 路徑編號
   */
  id: number;
  /**
   * 目的地名稱
   * @example ‘青青草原’ | ‘新竹體育館’
   */
  destination: string;
  /**
   * 本地位置名稱
   * @example ‘本地’ | '陽明交大'
   */
  location: string;
  /**
   * 耗時
   * @example "1h 30m"
   */
  costTime: string;
  /**
   * 到達時間
   * @example "11:30"
   */
  arrivalTime: string,
  /**
   * 交通工具種類以及%數
   */
  transportComp: Record<'公車' | 'Bike' | '步行', number>
  /**
   * 擁擠程度
   */
  crowding: number,
  /**
   * 子路徑選項
   */
  path_details: PathDetail[]
}